components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/BrandingPreferenceRetrievalClient.java

```diff
@@ -0,0 +1,177 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.mgt.endpoint.util.client;
+
+import org.apache.axis2.transport.http.HTTPConstants;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.log4j.Logger;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.wso2.carbon.identity.mgt.endpoint.util.IdentityManagementEndpointUtil;
+import org.wso2.carbon.identity.mgt.endpoint.util.IdentityManagementServiceUtil;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+
+/**
+ * Client to interact with the Tenant branding preferences API.
+ */
+public class BrandingPreferenceRetrievalClient {
+
+    private static final String CLIENT = "Client ";
+    private static final Logger log = Logger.getLogger(PreferenceRetrievalClient.class);
+    private static final String BRANDING_PREFERENCE_API_RELATIVE_PATH = "/api/server/v1/branding-preference";
+    private static final String RESOURCE_TYPE_URL_SEARCH_PARAM = "type";
+    private static final String RESOURCE_NAME_URL_SEARCH_PARAM = "name";
+    private static final String RESOURCE_LOCALE_URL_SEARCH_PARAM = "locale";
+
+    /**
+     * Check for branding preference in the given tenant.
+     *
+     * @param tenant Tenant Domain.
+     * @param type Resource Type. ex: ORG, APP, etc.
+     * @param name Resource Name. ex: Console, My Account, etc.
+     * @param locale ISO language code of the resource. ex: en-US, pt-BR, etc.
+     * @return A JSON Object containing branding preferences.
+     * @throws BrandingPreferenceRetrievalClientException
+     */
+    public JSONObject getPreference(String tenant, String type, String name, String locale)
+            throws BrandingPreferenceRetrievalClientException {
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
+
+            String uri = getBrandingPreferenceEndpoint(tenant);
+
+            try {
+                URIBuilder uriBuilder = new URIBuilder(uri);
+
+                if (StringUtils.isNotBlank(type)) {
+                    if (log.isDebugEnabled()) {
```

Comment:
Shall we improve the debug log a bit? we can try to add tenant, combine all the params in a single debug log WDYT?

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/3831/commits/e17df8230b25ece4ab75d5372a3464beeecb2543